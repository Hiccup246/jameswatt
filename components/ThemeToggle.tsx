import React from "react";
import MoonIcon from "./Icons/MoonIcon";
import SunIcon from "./Icons/SunIcon";
import { ThemeContext } from "./ThemeProvider";

function themeLabel(theme: string | undefined) {
  if (theme == undefined) return <></>;

  if (theme == "dark") {
    return "Light Mode";
  } else {
    return "Dark Mode";
  }
}

function themeIcon(theme: string | undefined) {
  if (theme == undefined) return <></>;

  if (theme == "dark") {
    return <SunIcon />;
  } else {
    return <MoonIcon />;
  }
}

export default function ThemeToggle() {
  const { siteTheme, flipTheme } = React.useContext(ThemeContext);

  if (!siteTheme) {
    return <></>;
  }

  return (
    <button
      className="hover group flex justify-center"
      onClick={() => flipTheme()}
    >
      <div className="mr-2 h-4 w-4">{themeIcon(siteTheme)}</div>

      <span className="group-hover:underline">{themeLabel(siteTheme)}</span>
    </button>
  );
}
