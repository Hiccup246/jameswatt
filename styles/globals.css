@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --slider-items: 7;
    --slider-item-sets: 2;
    --all-slider-items: calc(var(--slider-items) * var(--slider-item-sets));
    --slider-item-width: 200px;
  }

  html,
  body {
    padding: 0;
    margin: 0;
    font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
      "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif,
      "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol",
      "Noto Color Emoji";
    @apply bg-brown;
  }

  @property --gradient-angle {
    syntax: "<angle>";
    inherits: false;
    initial-value: 1turn;
  }
}

@layer utilities {
  .animation-pause {
    animation-play-state: paused;
  }

  .animation-play {
    animation-play-state: running;
  }
}
