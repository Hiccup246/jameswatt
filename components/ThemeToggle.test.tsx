import { expect, test } from "@jest/globals";
import ThemeToggle from "./ThemeToggle";
import renderer from "react-test-renderer";

test("renders the ThemeToggle correctly", () => {
  const tree = renderer.create(<ThemeToggle />).toJSON();

  expect(tree).toMatchSnapshot();
});
