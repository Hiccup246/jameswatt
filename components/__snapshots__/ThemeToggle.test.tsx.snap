// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ThemeToggle correctly 1`] = `
<button
  className="hover group flex justify-center"
  onClick={[Function]}
>
  <div
    className="mr-2 h-4 w-4"
  />
  <span
    className="group-hover:underline"
  />
</button>
`;
